### ISSUE TITLE
panicked at worker.rs:504:10:

### ISSUE THREAD

**Issue by OdenShirataki** [denoland/deno#21928]
Version: deno_runtime 0.139.0

Cargo.toml
```toml
[dependencies]
deno_runtime = { version = "0.139.0", features = ["dont_use_runtime_snapshot"] }
```

test.rs
```rust
let mut worker = MainWorker::bootstrap_from_options(
      deno_core::resolve_url("data:,").unwrap(),
      PermissionsContainer::allow_all(),
      WorkerOptions {
          module_loader: WdModuleLoader::new(cache_dir),
          ..Default::default()
      },
  );
```
When I run the above code it panics.

```error
deno_runtime-0.139.0\worker.rs:504:10:
called `Result::unwrap()` on an `Err` value: BadType { actual: "v8::data::Value", expected: "v8::data::Object" }
```

The relevant code is as follows.
```rust
      let global_obj = context_local.global(scope);
      let bootstrap_str =
        v8::String::new_external_onebyte_static(scope, b"bootstrap").unwrap();
      let bootstrap_ns: v8::Local<v8::Object> = global_obj
        .get(scope, bootstrap_str.into())
        .unwrap()
        .try_into()
        .unwrap();
```

I think it's because there is no object registered for global with the name "bootstrap", but I don't know how to register it before creating the MainWorker.

**Comment by tonywu6** [comment]
Looks like this might be related to #21137 ?

The examples under <https://github.com/denoland/deno/tree/main/runtime/examples> are also panicking at this location.

I found two configurations that work currently (with `deno_runtime 0.140.0`)

1. Enable features <code>["dont_use_runtime_snapshot", <strong>"__runtime_js_sources"</strong>]</code>

   I assume this means `deno_runtime` will fallback to compiling extensions from JS source which would then provide the expected `bootstrap` object

2. Compile the snapshot (during `build.rs`), and provide it through `WorkerOptions::startup_snapshot`. Pieced together using <https://deno.com/blog/roll-your-own-javascript-runtime-pt3>. Something like:

   In `build.rs`

   ```rs
   use deno_runtime::snapshot::create_runtime_snapshot;

   fn main() {
       create_runtime_snapshot("runtime.bin".into(), Default::default());
   }
   ```

   In `src/main.rs`

   ```rs
   static RUNTIME_SNAPSHOT: &[u8] = include_bytes!("../runtime.bin");

   #[tokio::main]
   async fn main() -> Result<(), AnyError> {
       // ...
       let mut worker = MainWorker::bootstrap_from_options(
           // ...
           WorkerOptions {
               startup_snapshot: Some(Snapshot::Static(RUNTIME_SNAPSHOT)),
               // ...
           },
       );
   }
   ```

Context: I'm trying to embed `deno_runtime`.

(cc @bartlomieju sorry for ping, complete noob here trying to learn Deno, I looked at the `create_runtime_snapshot` function, is there a specific step in it that provided the `bootstrap` object, that, when not included, causes the runtime to fail with the default configuration?)

**Comment by OdenShirataki** [comment]
@tonywu6 Thank you for answering.

After writing Cargo.toml as shown below, the panic no longer occurs.

`deno_runtime = { version = "0.140.0", features = [
    "dont_use_runtime_snapshot",
    "__runtime_js_sources",
] }`
(The same goes for 0.139.0)

However, I feel that the processing speed has decreased compared to 1.137 and earlier.

I think it takes a long time because the bootstrap snapshot is deleted from the runtime and the initial settings are executed without the snapshot, but I have not checked the source.


Therefore,

Cargo.toml
`deno_runtime = { version = "0.140.0" }`

and wrote it in the comments
> Compile the snapshot (during build.rs), and provide it through WorkerOptions::startup_snapshot.

I tried that and the problem was resolved.

However, the examples code(https://github.com/denoland/deno/tree/main/runtime/examples ) will panic, and it is unclear whether the speed reduction due to specification changes is what the developer intended, so I will leave it open.
